### PR TITLE
Added workaround to preprocess the new dataset

### PR DESCRIPTION
**Pull request by anushka255**

### CODE REVIEW

**Review by vineetbansal — COMMENTED** [review]
I don't see how this would work in the general case - if I remember correctly, some part of the code calls `read_file` on a csv/tsv and expects all columns to be returned. We shouldn't assume that just because we encountered a csv/tsv, the caller is only interested in the smiles. The `[1]` is problematic too (find the location of the smile column from the header and then index appropriately).

I would say:
- Determine if the file is a csv/tsv not by looking at the extension, but by looking at the first line.
- Find the position of the smile column
- Unless the caller has indicated otherwise, return the entire line
- If the caller wants only the smile (`smile_only=True`), return it
- Use `smile_only=True` in the place where you find you need it (your new preprocess step, for example).

This should cause the old code to not break.

File: src/clm/functions.py [hunk 1]
```diff
@@ -143,13 +143,22 @@ def read_file(smiles_file, max_lines=None):
     return lines_array
 
 
-def read_file_incremental(input_file, max_lines=None):
+def read_file_incremental(
+    input_file,
+    max_lines=None,
+    smile_only=True,
```
Comment (vineetbansal) [inline src/clm/functions.py:149]: The purpose of introducing this flag was to not have existing code break (where `read_file` is used to read the entire line - this is why we renamed the function from `read_smiles` to `read_file`). Thus, the default value of this should be `False` (or go through the entire code to find out where it was used and change the calling code to `smile_only=False`).

Further, the caller may just use `read_file` and not `read_file_incremental`. So this flag should be passed down from `read_file` to `read_file_incremental`.

File: src/clm/functions.py [hunk 2]
```diff
@@ -143,13 +143,22 @@ def read_file(smiles_file, max_lines=None):
...
+):
     count = 0
-    for line in open(input_file, "r").readlines():
-        yield line.strip()
-        count += 1
-        if max_lines is not None and count == max_lines:
-            return
+    with open(input_file, "r") as f:
+        # Extract index of smiles from header
+        if not smile_only:
```
Comment (vineetbansal) [inline src/clm/functions.py:154]: `smile_only` indicates what the caller wants this function to return, not the format of the file.

Comment (anushka255) [reply src/clm/functions.py]: Got it! Updated the usage of `smile_only` variable accordingly.

File: src/clm/functions.py [hunk 3]
```diff
@@ -137,19 +137,28 @@ def get_rdkit_fingerprints(mols, include_none=False):
...
+):
     count = 0
-    for line in open(input_file, "r").readlines():
-        yield line.strip()
-        count += 1
-        if max_lines is not None and count == max_lines:
-            return
+    with open(input_file, "r") as f:
+        # Extract index of smiles from header
+        if smile_only:
```
Comment (vineetbansal) [inline src/clm/functions.py:154]: Again, `smile_only` is an indication by the caller on what they want. It doesn't tell you whether the file contains smiles or it is a csv file.

**Pull request merged**